File: lang/go/test/muse/data_structures/ArrayStack/ArrayStackTest.go
package main

import (
    "muse/data_structures/ArrayStack"
    "muse/util/TestRunner"
)

func testArrayStack() bool {
    size := 8192

    stack := ArrayStack.New(0)

    for i := 1; i <= size; i++ {
        stack.Push(i)
    }

    stack.Shrink()

    if stack.Size() != size {
        return false
    }

    if stack.Capacity() != size {
        return false
    }

    for i := size; i > 0; i-- {
        if stack.Peek() != i {
            return false
        }
        stack.Pop()
    }

    stack.Shrink()

    if !stack.IsEmpty() {
        return false
    }

    return stack.Capacity() == 0
}

func main() {
    TestRunner.ParseTest(testArrayStack())
}

File: ruby/test/muse/data_structures/ArrayStackTest.rb
require "muse/data_structures/ArrayStack"
require "muse/util/TestRunner"

def testArrayStack()
    size = 8192

    stack = ArrayStack::ArrayStack.new(0)

    1.upto(size) do |i|
        stack.push(i)
    end

    stack.shrink()

    if stack.size() != size
        return false
    end

    if stack.capacity() != size
        return false
    end

    size.downto(1) do |i|
        if stack.peek() != i
            return false
        end
        stack.pop()
    end

    stack.shrink()

    unless stack.isEmpty()
        return false
    end

    stack.capacity().zero?
end

if __FILE__ == $PROGRAM_NAME
    TestRunner.parseTest(testArrayStack())
end
